tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -200,6 +201,38 @@ def test_compare_validation_metrics(self):
             assert not tuner.compare_validation_metrics(0.5, 0.5)
             assert tuner.compare_validation_metrics(0.1, 0.5)
 
+    @pytest.mark.parametrize('metric, eval_at_param, expected', [
+        ('auc', {'eval_at': 5}, 'auc'),
```

Comment:
(nits) According to [the document](https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters), `eval_at` parameter is used only with `ndcg` and `map` metrics. How about removing the test case used with an unexpected metric in the test cases for `eval_at`.

Reply:
> According to the document, eval_at parameter is used only with `ndcg` and `map` metrics.

Thank you for your feedback. You're right.
But, I added these test cases to ensure that `_metric_with_eval_at` returns the identical metric names if they are not `ndcg` or `map` because `_metric_with_eval_at` will be applied to all metrics ([here](https://github.com/optuna/optuna/pull/912/files#diff-0d8b1d930818af2b13939666bb2df3eaR145)).

Reply:
It makes sense! LGTM:+1: